app/controllers/admin/api/personal/access_tokens_controller.rb

```diff
@@ -0,0 +1,99 @@
+  ##~ op.parameters.add :name => "scopes", :defaultName => "scopes[]", :description => "Scope of the access token. URL-encoded array containing one or more of the possible values values. The possible values are, for a master user [\"account_management\", \"stats\"], and for a tenant user [\"finance\", \"account_management\", \"stats\"]", :dataType => "custom", :allowMultiple => true, :required => true, :paramType => "query"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  #
+  def create
+    respond_with current_user.access_tokens.create(access_token_params)
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/personal/access_tokens.json"
+  ##~ e.responseClass = "List[access_token]"
+  #
+  ##~ op = e.operations.add
+  ##~ op.httpMethod = "GET"
+  ##~ op.summary   = "Access Token List"
+  ##~ op.description = "Returns the list of access tokens of the user. If the parameter name is sent, it returns only those whose name contain the string of the param"
+  ##~ op.group = "access_token"
+  #
+  ##~ op.parameters.add :name => "name", :description => "Part of the name of the access token.", :dataType => "string", :required => false, :paramType => "query"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  #
+  def index
+    respond_with current_user.access_tokens.by_name(params[:name] || '')
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/personal/access_tokens/{id}.json"
+  ##~ e.responseClass = "access_token"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "DELETE"
+  ##~ op.summary    = "Access Token Delete"
+  ##~ op.description = "Deletes an access token."
+  ##~ op.group = "access_token"
+  #
+  ##~ op.parameters.add :name => "id", :description => "ID or value of the access token.", :dataType => "integer", :paramType => "path", :required => true
+  #
+  ##~ op.parameters.add @parameter_access_token
+  #
+  def destroy
+    token.destroy
+    respond_with token
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/personal/access_tokens/{id}.json"
+  ##~ e.responseClass = "access_token"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "GET"
+  ##~ op.summary    = "Access Token Read"
+  ##~ op.description = "Shows an access token."
+  ##~ op.group = "access_token"
+  #
+  ##~ op.parameters.add :name => "id", :description => "ID or value of the access token.", :dataType => "integer", :paramType => "path", :required => true
+  #
+  ##~ op.parameters.add @parameter_access_token
+  #
+  def show
+    respond_with token
+  end
+
+
+  private
+
+  def token
+    @token ||= begin
+      id_or_value = params[:id]
+      scoped_access_tokens = current_user.access_tokens
+      scoped_access_tokens.by_value(id_or_value) || scoped_access_tokens.find(id_or_value)
```

Comment:
`by_value` looks like it will return an scope (or collection), though the implementation does not 
https://github.com/3scale/porta/blob/daf83afa159af2ce0923155aee8f6740660b17fa/app/models/access_token.rb#L119-L123
That looks strange to me. Anyway we could probably renaming it to: 

```ruby
def first_by_value_or_id(id_or_value)
  where.has { value = id_or_value | id = id_or_value }.first
end
```

Reply:
In https://github.com/3scale/porta/pull/528 😄